### PR TITLE
Philip/ussd

### PR DESCRIPTION
**Pull request by PhilipWafula**
Not ready for merge.
Some functions necessary for state machine are not yet available.

### CODE REVIEW

File: test/unit/app/utils/test_kenya_state_machine.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+                             (initial_pin_confirmation_state, standard_user, "1212", "exit_pin_mismatch"),
+                             # start state transition tests
+                             (start_state, standard_user, "1", "send_enter_recipient"),
+                             (start_state, standard_user, "2", "account_management"),
+                             (start_state, standard_user, "3", "directory_listing"),
+                             (start_state, standard_user, "4", "exchange_token"),
+                             (start_state, standard_user, "5", "help"),
+                             (start_state, standard_user, "6", "exit_invalid_menu_option"),
+                             # send_enter_recipient state tests
+                             # (send_enter_recipient_state, standard_user, "0712345678", "send_token_amount"),
```
Comment (wangfowen) [inline test/unit/app/utils/test_kenya_state_machine.py]: i'd put TODOs around these so it's more clear which ones are comments vs things that need to be fixed later

File: app/server/utils/ussd/kenya_ussd_state_machine.py [hunk 2]
```diff
@@ -1,189 +1,548 @@
...
+    def is_authorized_pin(self, user_input):
+        pass
+
+    def is_blocked_pin(self, user_input):
+        pass
+
+    def is_valid_new_pin(self, user_input):
+        pass
+
+    def save_business_directory_info(self, user_input):
```
Comment (wangfowen) [inline app/server/utils/ussd/kenya_ussd_state_machine.py]: should put TODOs on top of these methods so we know we need to fix these

File: app/server/utils/ussd/kenya_ussd_state_machine.py [hunk 3]
```diff
@@ -1,189 +1,548 @@
...
+
+from server import db
+from server.models.user import User
+from server.models.ussd import UssdSession
+from server.utils.phone import proccess_phone_number, send_generic_message
+
+
+def get_user(query_filter):
+    user = User.query.filter_by(phone=proccess_phone_number(phone_number=query_filter, region="KE")).first()
+    if user is not None:
```
Comment (wangfowen) [inline app/server/utils/ussd/kenya_ussd_state_machine.py]: what happens if user is None then?

File: test/unit/app/utils/test_kenya_state_machine.py (hunk 1 above)
Comment (PhilipWafula) [reply test/unit/app/utils/test_kenya_state_machine.py]: I'll add the TODOs on top of these and push

File: app/server/utils/ussd/kenya_ussd_state_machine.py (hunk 3 above)
Comment (PhilipWafula) [reply app/server/utils/ussd/kenya_ussd_state_machine.py]: Should have had an exception there, I'll add one.

**Pull request merged**